record.c

```diff
@@ -205,17 +210,43 @@ int janus_recorder_save_frame(janus_recorder *recorder, char *buffer, uint lengt
 		/* Done */
 		recorder->header = 1;
 	}
+	if(recorder->pause) {
+		if(!recorder->paused_ts) {
+			if(recorder->type == JANUS_RECORDER_DATA) {
+				recorder->paused_ts = janus_get_real_time();
+			} else {
+				recorder->paused_ts = ntohl(rtp->timestamp);
+			}
+			JANUS_LOG(LOG_INFO, "PAUSED TIME TS = %u, OFFSET=%u \n", recorder->paused_ts, recorder->offset_ts);
+		}
+		janus_mutex_unlock_nodebug(&recorder->mutex);
+		return 0;
+	} else if(recorder->paused_ts) {
+		if(recorder->type == JANUS_RECORDER_DATA) {
+			recorder->offset_ts += janus_get_real_time() - recorder->paused_ts;
+		} else {
+			recorder->offset_ts = (uint32_t) recorder->offset_ts + (ntohl(rtp->timestamp) - (uint32_t)recorder->paused_ts);
+			JANUS_LOG(LOG_INFO, "RESUME TIME TS = %u OFFSET=%u \n", rtp->timestamp, recorder->paused_ts, recorder->offset_ts);
+		}
+		recorder->paused_ts = 0;
```

Comment:
What happens here when you pause/resume more than once? Not sure this would still work. Besides, there's no fix to the sequence numbers, which means they won't be sequential, with big holes in between. This will cause severe issues when trying to replay them via Record&Play, because the recipient will think packets have been missed, and NACK them. This is one of the reasons why I'm insisting on using the RTP context here too, as it's one of the things it takes care of for you.